localstack-core/localstack/services/secretsmanager/provider.py

```diff
@@ -746,15 +750,17 @@ def backend_rotate_secret(
             if rotation_period < 1 or rotation_period > 1000:
                 msg = "RotationRules.AutomaticallyAfterDays must be within 1-1000."
                 raise InvalidParameterException(msg)
+    else:
+        # Resolve auto_rotate_after_days and fallback to previous value
+        # if its missing from the current request
+        rotation_period = secret.auto_rotate_after_days
```

Comment:
this could cause 'rotation_period' variable to be referenced before assignment [here](https://github.com/mabuaisha/localstack/blob/secret-manager-rotation-secret-issue/localstack-core/localstack/services/secretsmanager/provider.py#L791)

Reply:
Yeah thats right, I will init `rotation_period = 0`

Reply:
Is it always `0` or can we move the `rotation_period = secret.auto_rotate_after_days` above the `if` block and have _that_ as the default, then remove the `else` branch?

Reply:
I can remove the `else` branch and make it the default one. However, I will fallback to `0` like this

    rotation_period = secret.auto_rotate_after_days or 0

That way even if previous value for for `auto_rotate_after_days` that is part of secret is not integer, we are make sure always if the value is not set the default one is `0`